app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/MapDataWithEditsSource.kt

```diff
@@ -438,3 +464,6 @@ class MapDataWithEditsSource internal constructor(
         listeners.forEach { it.onCleared() }
     }
 }
+
+private fun Element.isEqualExceptVersionAndTimestamp(element: Element?): Boolean =
+    element?.copy(timestampEdited = timestampEdited, version = version) == this
```

Comment:
it's unexpected (and certainly slower) to copy an element for a comparison. I'd very much prefer something like
```kotlin
/** returns whether this element equals given [element], ignoring any versioning info */
private fun Element.equalsInData(element: Element): Boolean = 
  id == element.id && tags == element.tags && when {
  is Node -> position == element.position // (or .equalsInOsm??)
  is Way -> ...etc
  is Relation ->...etc
}

```

 even if it is a few more lines of code